python_modules/libraries/dagster-databricks/dagster_databricks/databricks.py

```diff
@@ -564,7 +564,7 @@ def submit_run(self, run_config: Mapping[str, Any], task: Mapping[str, Any]) ->
                 new_cluster["instance_pool_id"] = nodes["instance_pool_id"]
 
                 if "driver_instance_pool_id" in nodes:
-                    new_cluster["driver_node_type_id"] = nodes["driver_node_type_id"]
+                    new_cluster["driver_instance_pool_id"] = nodes["driver_instance_pool_id"]
```

Comment:
@smackesey  can you take a look at this PR? it looks like it's potentially a copy-paste issue, but it's unclear to me if we should be doing 
```python
if "driver_instance_pool_id" in nodes:
                    new_cluster["driver_instance_pool_id"] = nodes["driver_instance_pool_id"]
```

or 
```python
if "driver_node_type_id" in nodes:
                    new_cluster["driver_node_type_id"] = nodes["driver_node_type_id"]
```

or both

Reply:
I can try to shed some additional light on the issue. Also, happy to provide additional details.

Here is the error that is being raised when attempting to set `driver_instance_pool_id`.
<img width="1522" alt="Screenshot 2024-02-28 at 8 49 36 PM" src="https://github.com/dagster-io/dagster/assets/23409221/d1b5b653-7b0c-4361-9c8e-55d058a26a9c">

The code is currently attempting to access `driver_node_type_id` from `cluster.new.nodes`. However, the step launcher's `run_config` spec states that the `cluster.new.nodes` field expects `driver_node_type_id`  and `node_type_id`  to be nested within an object called `node_types`. The fields `driver_instance_pool_id` and `instance_pool_id` on the other hand are first-class properties alongside `node_types` (see run_config spec [here](https://github.com/dagster-io/dagster/blob/2cabe8733cb517c8caaa21e0b323f46b944ef3ef/python_modules/libraries/dagster-databricks/dagster_databricks/configs.py#L362)). Later on [here](https://github.com/dagster-io/dagster/blob/2cabe8733cb517c8caaa21e0b323f46b944ef3ef/python_modules/libraries/dagster-databricks/dagster_databricks/databricks.py#L572), `driver_node_type_id` is accessed from `node_types` per the `run_config` spec.

Reply:
Any thoughts or concerns here, @smackesey? We are hoping to leveraging this feature of databricks in our dagster deployment to reduce DBX cluster startup time and improve job runtime.

Reply:
Hi Steven, sorry for the delay here, I am running CI and if all is well I will land it and it will go out in this week's release.

Reply:
No problem at all! This will be huge for us - thank you :).